universe/Universe.cpp

```diff
@@ -83,345 +84,11 @@ namespace boost {
-    {
-        typedef boost::on_finish_vertex event_filter;
-
-        struct FoundDestination {}; // exception type thrown when destination is found
-
-        PathFindingShortCircuitingVisitor(int dest_system) : destination_system(dest_system) {}
-        template <class Vertex, class Graph>
-        void operator()(Vertex u, Graph& g)
-        {
-            if (static_cast<int>(u) == destination_system)
-                throw FoundDestination();
-        }
-        const int destination_system;
-    };
-
-    /** Complete BFS visitor implementing:
-      *  - predecessor recording
-      *  - short-circuit exit on found match
-      *  - maximum search depth 
-      */
-    template <class Graph, class Edge, class Vertex> class BFSVisitorImpl
-    {
-    public:
-        class FoundDestination {}; 
-        class ReachedDepthLimit {};
-
-    private:
-        Vertex m_marker;
-        Vertex m_stop;
-        Vertex m_source;
-        Vertex * m_predecessors;
-        int m_levels_remaining;
-        bool m_level_complete;
-
-    public:
-        BFSVisitorImpl(const Vertex& start, const Vertex& stop, Vertex predecessors[], int max_depth)
-            : m_marker(start),
-              m_stop(stop),
-              m_source(start),
-              m_predecessors(predecessors),
-              m_levels_remaining(max_depth),
-              m_level_complete(false)
-        {}
-
-        void initialize_vertex(const Vertex& v, const Graph& g)
-        {}
-
-        void discover_vertex(const Vertex& v, const Graph& g) {
-            m_predecessors[static_cast<int>(v)] = m_source;
-
-            if (v == m_stop)
-                throw FoundDestination();
-
-            if (m_level_complete) {
-                m_marker = v;
-                m_level_complete = false;
-            }
-        }
-
-        void examine_vertex(const Vertex& v, const Graph& g) {
-            if (v == m_marker) {
-                if (!m_levels_remaining)
-                    throw ReachedDepthLimit();
-                m_levels_remaining--;
-                m_level_complete = true;
-            }
-            
-            m_source = v; // avoid re-calculating source from edge
-        }
-
-        void examine_edge(const Edge& e, const Graph& g) {}
-        void tree_edge(const Edge& e, const Graph& g) {}    // wait till target is calculated
-
-
-        void non_tree_edge(const Edge& e, const Graph& g) {}
-        void gray_target(const Edge& e, const Graph& g) {}
-        void black_target(const Edge& e, const Graph& g) {}
-        void finish_vertex(const Vertex& e, const Graph& g) {}
-    };
-
-    ////////////////////////////////////////////////////////////////
-    // templated implementations of Universe graph search methods //
-    ////////////////////////////////////////////////////////////////
-    struct vertex_system_id_t {typedef boost::vertex_property_tag kind;}; ///< a system graph property map type
```

Comment:
presumably a header or three can be removed from Universe.cpp if uses of this sort of type are removed from the file...